### PR TITLE
[1801] Add information to course sync errors

### PR DESCRIPTION
**Pull request by defong**
### Context
Usefully useless error message

### Changes proposed in this pull request
Usable error message 

### Guidance to review
:shipit: 
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[When a course is unsyncable then the exception raised needs to show the reasons](https://trello.com/c/wW0RBoiV/1801-when-a-course-is-unsyncable-then-the-exception-raised-needs-to-show-the-reasons)

File: spec/requests/api/v2/providers/courses/publish_spec.rb [hunk 1]
```diff
@@ -109,7 +109,7 @@
           expect {
             subject
           }.to raise_error RuntimeError,
-                           'course is not syncable'
+                           'course ' + "'#{course}'" + ' is not syncable due to: {:dfe_subjects=>[{:error=>"No DfE subject."}]}'
```
Comment (faissalMT) [inline spec/requests/api/v2/providers/courses/publish_spec.rb]:
```suggestion
                           "course '#{course}' is not syncable due to: {:dfe_subjects=>[{:error=>\"No DfE subject.\"}]}"
```

File: app/controllers/api/v2/courses_controller.rb [hunk 2]
```diff
@@ -138,7 +138,7 @@ def has_synced?
           SearchAndCompareAPIService::Request.sync([@course])
         else
           raise RuntimeError.new(
-            'course is not syncable'
+            "course '#{@course}' is not syncable due to: #{@course.errors.details}"
```
Comment (dankmitchell) [inline app/controllers/api/v2/courses_controller.rb]: `"#{@course.name} (#{@course.code}) sync error: #{@course.errors.details}"` ?

Comment (defong) [reply app/controllers/api/v2/courses_controller.rb]: How about with provider
```ruby
"'#{@course}' '#{@course.provider}' sync error: #{@course.errors.details}" 
```

it gives
```ruby
# it gives
course_name (course_code) provider_name (provider_code) ....
'G (C3D3)' 'ACME SCITT28921 (A1)' sync error: {:dfe_subjects=>[{:error=>"No DfE subject."}]}
```

**Pull request merged**